rust/src/tx_builder.rs

```diff
@@ -190,18 +190,23 @@ impl TransactionBuilderConfigBuilder {
         }
     }
 
-    pub fn fee_algo(mut self, fee_algo: fees::LinearFee) -> Self {
-        self.fee_algo = Some(fee_algo);
+    pub fn fee_algo(mut self, fee_algo: &fees::LinearFee) -> Self {
+        self.fee_algo = Some(fee_algo.clone());
         self
     }
 
-    pub fn pool_deposit(mut self, pool_deposit: BigNum) -> Self {
-        self.pool_deposit = Some(pool_deposit);
+    pub fn coins_per_utxo_word(mut self, coins_per_utxo_word: &Coin) -> Self {
+        self.coins_per_utxo_word = Some(coins_per_utxo_word.clone());
         self
     }
 
-    pub fn key_deposit(mut self, key_deposit: BigNum) -> Self {
-        self.key_deposit = Some(key_deposit);
+    pub fn pool_deposit(mut self, pool_deposit: &BigNum) -> Self {
```

Comment:
I'm fairly certain this (`BigNum` passed by-value) is present in other areas. I know it implements `Copy` in rust but I'm not sure if that is enough to not cause issues when it goes through the wasm-bindgen boundary.

Reply:
For me it caused problem when being passed by value - that value then gets very counter-intuitively erased when used from JS. I will double-check then where it's being used by-value, might also need to be fixed if this is something that is used from JS as well.